presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -745,15 +751,19 @@ public synchronized void addPartition(
         }
     }
 
-    public synchronized void dropPartition(ConnectorSession session, String databaseName, String tableName, List<String> partitionValues)
+    public synchronized void dropPartition(ConnectorSession session, String databaseName, String tableName, List<String> partitionValues, boolean dropData)
```

Comment:
There are many changes here which are related to adding `dropData` flag.
This makes it hard to follow.

Are you positive they are all necessary?

Anyway, it would be best to separate them into a prep commit (which should be easy -- generally everything except the two new procedures). This would allow giving the changes a name that explains the change _here_.

Reply:
I've separated commits.

I'm positive that this changes are required.